website/docs/abstract.md

```diff
@@ -122,6 +122,157 @@ As the example shows, there are two main steps:
+
+- The `IFoo` module declares a single, abstract `foo` method. All classes that
+  include this module must either be marked abstract or define this method.
+
+- The `Parent` method does **not** depend on `IFoo`, **but** does happen to
+  define a method called `foo`.
+
+- Both `Child` and `NotAParent` have `include IFoo`, but neither define a `foo`
+  method.
+
+- Despite this: only `NotAParent` has an error saying that a concrete
+  implementation of `foo` is missing. Sorbet allows the `foo` method to be
+  implemented in `Child` because it inherits a `foo` method from `Parent`.
+
+### Approximating duck types
+
+This technique is particularly useful as a way to approximate "duck typing,"
+where you depend on "anything type, so long as it has this method."
+
+For example:
+
+```ruby
+module ShortName
+  extend T::Helpers
+  abstract!
+  sig { abstract.returns(T.nilable(String)) }
+  def name; end
+
+  sig { returns(T.nilable(String)) }
+  def short_name
+    self.name&.split('::')&.last
+  end
+end
+```
+
+This module provides a `short_name` method (which computes the "short name" of a
+something like a `Module` by splitting the full name into `::`-delimited tokens
+and returning the last one. Like `C` for `module A::B::C`).
+
+The module's implementation depends on the `name` method existing. If we don't
+declare it as an `abstract` method, Sorbet reports an error saying "Method
+`name` does not exist," which is true--there's no guarantee someone mixes this
+module into a context where `name` is defined.
+
+But by declaring `name` as an abstract method, Sorbet will check this property.
+In particular, this has the effect of catching someone who accidentally uses
+`include` instead of `extend` when mixing this module into a class:
+
+```ruby
+class A
+  include ShortName # ❌ error: Must define abstract method `name`
+end
+
+class B
+  extend ShortName # ✅
+end
+```
+
+This technique is particularly effective when it's not possible to refactor some
+upstream dependency's code to expose an explicit interface. The `Module` class
+in the Ruby stdlib doesn't have some sort of public `INameable` interface with
+the `name` method. A handful of database model classes in an application might
+share a set of related fields, without explicitly implementing some interface.
+And yet, using this technique Sorbet allows writing modules which depend on
+those implicit interfaces.
+
+This technique is also quite flexible: the `ShortName` module can be used in
+**any** context where a `name` method is available. So for example, if you had
+some `T::Struct` that stores a `name`, this `ShortName` mixin could also be
+used:
+
+```ruby
+class C < T::Struct
+  include ShortName
+  prop :name, String
+end
+
+C.new(name: "Some::Long::Namespace").short_name # => "Namespace"
+```
+
+### Declaring requirements of an interface
+
+Letting abstract methods be implemented by inherited methods is particularly
+useful for expressing dependencies between
```

Comment:
Readers everywhere would like to have this tension resolved, and how to express their dependencies properly.

Reply:
oops, I had auto merge on. I will fix